client/allocrunner/csi_hook.go

```diff
@@ -88,7 +90,20 @@ func (c *csiHook) claimVolumesFromAlloc() (map[string]*volumeAndRequest, error)
 
 	// Initially, populate the result map with all of the requests
 	for alias, volumeRequest := range tg.Volumes {
+
 		if volumeRequest.Type == structs.VolumeTypeCSI {
+
+			for _, task := range tg.Tasks {
+				caps, err := c.ar.GetTaskDriverCapabilities(task.Name)
+				if err != nil {
+					return nil, fmt.Errorf("could not validate task driver capabilities: %v", err)
+				}
+
+				if caps.MountConfigs == drivers.MountConfigSupportNone {
+					return nil, fmt.Errorf("task driver does not support CSI")
```

Comment:
Should we indicate which task or the task driver names in these error messages?

The `volume_hook.go` references will include the task names when logged hopefully, but not sure about this one....

Reply:
👍 Definitely some room for improvement. Will fix.